fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/NavigationBar.kt

```diff
@@ -17,166 +16,320 @@ import androidx.compose.material.IconButton
-                    ItemType.STANDARD -> {
-                        IconButton(onClick = {}) {
-                            Icon(
-                                painter = painterResource(it.iconId),
-                                stringResource(id = it.descriptionResourceId),
-                                tint = FirefoxTheme.colors.iconPrimary,
-                            )
-                        }
-                    }
-                    ItemType.TAB_COUNTER -> {
-                        CompositionLocalProvider(LocalContentColor provides FirefoxTheme.colors.iconPrimary) {
-                            IconButton(onClick = {}) {
-                                TabCounter(tabCount = tabCount)
-                            }
-                        }
-                    }
-
-                    ItemType.MENU -> {
-                        // [ActionItem] will be refactored here:
-                        // https://bugzilla.mozilla.org/show_bug.cgi?id=1878827
-                        if (menuButton != null) {
-                            AndroidView(
-                                modifier = Modifier.size(48.dp),
-                                factory = { _ -> menuButton },
-                            )
-                        } else {
-                            IconButton(onClick = {}) {
-                                Icon(
-                                    painter = painterResource(it.iconId),
-                                    stringResource(id = it.descriptionResourceId),
-                                    tint = FirefoxTheme.colors.iconPrimary,
-                                )
-                            }
-                        }
-                    }
-                }
-            }
-        }
+        BackButton(
+            onBackButtonClick = onBackButtonClick,
+            onBackButtonLongPress = onBackButtonLongPress,
+        )
+
+        ForwardButton(
+            onForwardButtonClick = onForwardButtonClick,
+            onForwardButtonLongPress = onForwardButtonLongPress,
+        )
+
+        HomeButton(
+            onHomeButtonClick = onHomeButtonClick,
+        )
+
+        TabsButton(
+            onTabsButtonClick = onTabsButtonClick,
+            tabCount = tabCount,
+        )
+
+        MenuButton(menuButton = menuButton)
     }
 }
 
 /**
- * Represents a navigation bar element.
+ * Top-level UI for displaying the navigation bar.
  *
- * @property iconId Resource ID of the icon that item should display.
- * @property descriptionResourceId Text used as a content description by accessibility services.
- * @property type Type of the item, defaults to [ItemType.STANDARD].
+ * @param onSearchButtonClick
+ * @param menuButton
+ * @param onTabsButtonClick
+ * @param isPrivateMode
+ * @param browserStore
  */
-data class ActionItem(
-    val iconId: Int,
-    val descriptionResourceId: Int,
-    val type: ItemType = ItemType.STANDARD,
-)
+@Composable
+fun HomeNavBar(
```

Comment:
Looking at the two nav bar composables, maybe we can consolidate them slightly into something like:

```
@Composable
private fun NavBar(
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    Row(
        modifier = Modifier
            .background(FirefoxTheme.colors.layer1)
            .height(48.dp)
            .fillMaxWidth()
            .padding(horizontal = 16.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
        content = content,
    )
}

@Composable
fun HomeNavBar(
    ...
) {
    NavBar {
        // Button composables 
    }
}
```
This will help mitigate any design regressions for at least the Nav bar container